server/core/src/main/java/org/infinispan/server/core/configuration/ProtocolServerConfigurationBuilder.java

```diff
@@ -102,58 +109,55 @@ public SslConfigurationBuilder ssl() {
       return ssl;
    }
 
+   @Override
+   public S ioThreads(int ioThreads) {
+      attributes.attribute(IO_THREADS).set(ioThreads);
+      return this.self();
+   }
+
    @Override
    public S workerThreads(int workerThreads) {
-      this.workerThreads = workerThreads;
+      attributes.attribute(WORKER_THREADS).set(workerThreads);
       return this.self();
    }
 
    @Override
    public S startTransport(boolean startTransport) {
-      this.startTransport = startTransport;
+      attributes.attribute(START_TRANSPORT).set(startTransport);
       return this.self();
    }
 
    @Override
    public S adminOperationsHandler(AdminOperationsHandler handler) {
-      this.adminOperationsHandler = handler;
+      attributes.attribute(ADMIN_OPERATION_HANDLER).set(handler);
       return this.self();
    }
 
 
    @Override
    public void validate() {
       ssl.validate();
-      if (idleTimeout < -1) {
-         throw log.illegalIdleTimeout(idleTimeout);
+      if (attributes.attribute(IDLE_TIMEOUT).get() < -1) {
+         throw log.illegalIdleTimeout(attributes.attribute(IDLE_TIMEOUT).get());
+      }
+      if (attributes.attribute(SEND_BUF_SIZE).get() < 0) {
+         throw log.illegalSendBufferSize(attributes.attribute(SEND_BUF_SIZE).get());
       }
-      if (sendBufSize < 0) {
-         throw log.illegalSendBufferSize(sendBufSize);
+      if (attributes.attribute(RECV_BUF_SIZE).get() < 0) {
+         throw log.illegalReceiveBufferSize(attributes.attribute(RECV_BUF_SIZE).get());
       }
-      if (recvBufSize < 0) {
-         throw log.illegalReceiveBufferSize(recvBufSize);
+      if (attributes.attribute(IO_THREADS).get() < 0) {
```

Comment:
It seems to me that `0` should be illegal value, too, since it does not retrigger default settings (based on CPU count). Same for worker threads.

Reply:
0 is perfectly legal for netty slaves https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/MultithreadEventLoopGroup.java#L52

Reply:
As for documenting properties, my opinion is that we should but only for those properties which are not exposed in the configuration, because there we support property expansion anyway.
I was not happy about keeping that workerThreads property either, but did so because of backwards compatibility. Maybe we should list all such properties as deprecated so that we can remove them  in Infinispan 10?

Reply:
If it's not documented, there's no place to deprecate. We could remove that in 10 without warning.